api/base/serializers.py

```diff
@@ -385,12 +394,23 @@ def get_url(self, obj, view_name, request, format):
                 if kwargs is None:
                     urls[view_name] = {}
                 else:
-                    urls[view_name] = self.reverse(view, kwargs=kwargs, request=request, format=format)
-
+                    url = self.reverse(view, kwargs=kwargs, request=request, format=format)
+                    if self.filter:
+                        url += self.format_filter(obj)
+                    urls[view_name] = url
         if not urls['self'] and not urls['related']:
             urls = None
         return urls
 
+    def format_filter(self, obj):
```

Comment:
Just in the spirit of not reinventing the wheel, this might be a easier route to the same outcome. Plus, someone already wrote tests for it so you don't have to.

https://docs.djangoproject.com/en/1.9/ref/request-response/#django.http.QueryDict.update

QueryDicts are immutable by default but you can pass a mutable kwarg to them and then `qd.update(regular_dictionary)` will work.

Then `qd.urlencode()` will get you a properly formatted querystring that's urlencoded. You don't have to worry about the last & or any of that stuff just `"{}?{}".format(the_url, qd.urlencode())`